src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/mobile/close_application_response_test.cc

```diff
@@ -0,0 +1,117 @@
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <memory>
+
+#include "mobile/close_application_response.h"
+#include "mobile/close_application_request.h"
+
+#include "gtest/gtest.h"
+
+#include "smart_objects/smart_object.h"
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/mock_event_dispatcher.h"
+
+namespace test {
+namespace components {
+namespace command_test {
+namespace mobile_command_test {
+namespace close_application_response {
+
+using ::testing::_;
+using ::testing::Return;
+using ::testing::SaveArg;
+using ::testing::DoAll;
+using am::commands::MessageSharedPtr;
+namespace am = ::application_manager;
+
+typedef std::shared_ptr<sdl_rpc_plugin::commands::CloseApplicationResponse>
+    CloseApplicationCommandPtr;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+const uint32_t kCorrelationId = 2u;
+}
+
+MessageSharedPtr CreateCloseAppMessage() {
+  using namespace am;
+  auto message = std::make_shared<smart_objects::SmartObject>(
+      smart_objects::SmartType_Null);
+  auto& params = (*message)[strings::params];
+  params[strings::connection_key] = kConnectionKey;
+  params[strings::function_id] = mobile_apis::FunctionID::CloseApplicationID;
+  return message;
+}
+
+class CloseApplicationResponseTest
+    : public commands_test::CommandsTest<
+          commands_test::CommandsTestMocks::kIsNice> {
+ public:
+  CloseApplicationResponseTest()
+      : message_(CreateCloseAppMessage())
+      , command_(
+            CreateCommand<sdl_rpc_plugin::commands::CloseApplicationResponse>(
+                message_)) {}
+
+  MessageSharedPtr message_;
+  CloseApplicationCommandPtr command_;
+};
+
+TEST_F(CloseApplicationResponseTest, CloseApplicationResponse_SUCCESS) {
```

Comment:
@IGapchuk Should we check not only SUCCCESS case , what about negative case?

Reply:
@ZhdanovP We check UNSUCCESS cases above this case.